tests/PHPStan/Rules/Variables/DefinedVariableRuleTest.php

```diff
@@ -102,6 +102,10 @@ public function testDefinedVariables(): void
 				'Undefined variable: $variableInEmpty',
 				145,
 			],
+			[
+				'Undefined variable: $negatedVariableInEmpty',
```

Comment:
@ondrejmirtes one last thing I am wondering:

https://phpstan.org/r/60a66a37-7ea8-45d6-a0db-3de562d5b88d

before this PR a variable in `!empty()` had a YES but in `isset()` we kept the NO certainty.
since you said a NO can't get into YES and vice-versa, I think this was a bug before, and the error we are not getting here is expected.